USca/USca-Server/Tags/TagTrendingWorker.cs

```diff
@@ -0,0 +1,175 @@
+        {
+            while (Ws.State == WebSocketState.Open)
+            {
+                var buffer = new byte[1024 * 4];
+
+                try
+                {
+                    var result = await Ws.ReceiveAsync(buffer, CancellationToken.None);
+
+                    if (result.MessageType == WebSocketMessageType.Close)
+                    {
+                        await Ws.CloseAsync(WebSocketCloseStatus.NormalClosure, null, CancellationToken.None);
+                    }
+                    else
+                    {
+                        Console.WriteLine(Encoding.ASCII.GetString(buffer, 0, result.Count));
+                    }
+                }
+                catch (WebSocketException)
+                {
+                    break;
+                }
+            }
+        }
+
+        private void StartTagSync()
+        {
+            _tagSyncThread = new(new ThreadStart(TagSync));
+            _tagSyncThread.Start();
+        }
+
+        private void EndTagSync()
+        {
+            _tagSyncThread.Abort();
+        }
+
+        /// <summary>
+        /// This method syncs the tags from the database onto _threads.
+        /// This method should run in the tag sync thread.
+        /// </summary>
+        private void TagSync()
+        {
+            Thread.Sleep(1000);
+            Console.WriteLine("Sync tags");
+
+            var currentTags = _tagService.GetAll()
+                .Where(t => t.Mode == TagMode.Input)
+                .Select(t => t.Id)
+                .ToList();
+            var staleTags = _threads.Keys.ToList().Except(currentTags);
+
+            // Remove threads for tags that don't exist anymore.
+
+            foreach (var tagId in staleTags)
+            {
+                _threads[tagId].Abort();
+                _threads.Remove(tagId);
+                Console.WriteLine($"Removed thread for tag {tagId}.");
+            }
+
+            // Create threads for new tags.
+
+            foreach (var tagId in currentTags)
+            {
+                if (!_threads.ContainsKey(tagId))
+                {
+                    Tag? tag = _tagService.Get(tagId);
+                    if (tag == null)
+                    {
+                        // Should never happen, but VS complains about null checks.
+                        continue;
+                    }
+
+                    LoopThread rt = new(new ThreadStart(() => SendTagData(tag)));
+                    rt.Start();
+                    _threads[tagId] = rt;
+
+                    Console.WriteLine($"Added thread for tag {tagId}.");
+                }
+            }
+        }
+
+        /// <summary>
+        /// This method tries to send current data from the tag <c>t</c> across the WebSocket.
+        /// This method should run in a <c>LoopThread</c> assigned to the given tag.
+        /// </summary>
+        private void SendTagData(Tag t)
+        {
+            Thread.Sleep(t.ScanTime);
```

Comment:
As you mentioned, there will be an issue if the tag's `ScanTime` gets changed through `DbManager`, since we're using a local copy of the tag that doesn't reflect these changes. If you have a good idea on how to solve this go ahead, otherwise make an issue and someone will get to it.

Reply:
It's not required of us to add support for editing existing tags, so we can simply disable the existing edit functionality in DbManager. In case someone is interested in implementing things *with* editing, I'll make an issue but as far as I'm concerned, we don't have to do more work than what is needed.